components/post_view/embedded_bindings/button_binding/button_binding.tsx

```diff
@@ -3,25 +3,27 @@
 
 import React from 'react';
 
-import {AppBinding, AppCall, AppCallResponse} from 'mattermost-redux/types/apps';
+import {AppBinding, AppCallRequest, AppCallResponse, AppCallType} from 'mattermost-redux/types/apps';
 import {ActionResult} from 'mattermost-redux/types/actions';
-import {AppBindingLocations, AppCallResponseTypes, AppExpandLevels} from 'mattermost-redux/constants/apps';
+import {AppBindingLocations, AppCallResponseTypes, AppCallTypes, AppExpandLevels} from 'mattermost-redux/constants/apps';
 import {Post} from 'mattermost-redux/types/posts';
 
 import {injectIntl, IntlShape} from 'react-intl';
 
+import {Channel} from 'mattermost-redux/types/channels';
+
 import Markdown from 'components/markdown';
 import LoadingWrapper from 'components/widgets/loading/loading_wrapper';
-import {createCallRequest} from 'utils/apps';
+import {createCallContext, createCallRequest} from 'utils/apps';
 import {sendEphemeralPost} from 'actions/global_actions';
 
 type Props = {
     intl: IntlShape;
     binding: AppBinding;
-    userId: string;
     post: Post;
     actions: {
-        doAppCall: (call: AppCall) => Promise<ActionResult>;
+        doAppCall: (call: AppCallRequest, type: AppCallType) => Promise<ActionResult>;
+        getChannel: (channelId: string) => Promise<ActionResult>;
```

Comment:
is this part of the "minor refactoring" you refer to in the PR description?

Reply:
I have mixed feelings with this. What we were doing was getting the current team id. What we are doing now, is getting the channel where the post is to get the team id of the post.

At the moment, you cannot be in a thread of a different team. But if that was to change at any moment, this approach is more robust.

Nevertheless, I can go back to the previous approach.